beacon-chain/slasher/helpers.go

```diff
@@ -47,6 +46,55 @@ func (s *Service) groupByChunkIndex(
 	return attestationsByChunkIndex
 }
 
+// Validates the attestation data integrity, ensuring we have no nil values for
+// source, epoch, and that the source epoch of the attestation must be less than
+// the target epoch, which is a precondition for performing slashing detection.
+// This function also checks the attestation source epoch is within the history size
+// we keep track of for slashing detection.
+// This function returns a list of valid attestations, a list of attestations that are
+// valid in the future, and the number of attestations dropped.
+func (s *Service) validateAttestationIntegrity(
```

Comment:
Im not a fan of doing filtering and returning processed items in a func that beings with `validate` 

A better way would be filter this into `validateAttestationIntegrity` and `filterAttestation`